scripts/updateExtractions.js

```diff
@@ -0,0 +1,147 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+"use strict";
+
+const fs = require("fs");
+const path = require("path");
+const { execFileSync } = require("child_process");
+const yargs = require("yargs");
+const argv = yargs(process.argv).argv;
+
+// parse args
+const isCheck = "check" in argv;
+const targets = argv.targets;
+if (!targets) {
+  console.error(
+    `Fail! Please specify the "target" argument as a comma-separated list of paths to the target directories and files where extraction insertions need to be made.`,
+  );
+  process.exit(1);
+}
+
+// gather extractions from different packages into the workspace root
+console.log(`Gathering extractions from different packages...`);
+execFileSync("node", [path.join(__dirname, "gatherDocs.js")], { stdio: "inherit", cwd: path.join(__dirname, "..") });
+
+// set up constants
+const extractionsDir = path.join(__dirname, "..", "build/docs/extract");
+const extractionStart = "<!-- BEGIN EXTRACTION -->";
+const extractionEnd = "<!-- END EXTRACTION -->";
+const targetFileExtensions = [".ts", ".tsx", ".md"];
+const re = /^(\s*)(<!--|\/\/|\/\*)\s*\[\[include:\s*([\w\d\._-]+)(,[\s]*([\w\d_]+))?\]\]/;
```

Comment:
You can mark groups that you don't want captured `(?: ...)`, then capture indices would go in order:
```ts
const reIndentIndex = 1;
const reExtractionNameIndex = 2;
const reExtractionTypeIndex = 3;
```

Reply:
good point, applied the change